Foundation/Operation.swift

```diff
@@ -234,81 +234,163 @@ extension OperationQueue {
     public static let defaultMaxConcurrentOperationCount: Int = Int.max
 }
 
+internal class _IndexedOperationLinkedList {
+    
+    internal class Node {
+        var operation: Operation
+        var next: _IndexedOperationLinkedList.Node? = nil
+        weak var previous: _IndexedOperationLinkedList.Node? = nil
+        
+        init(operation: Operation) {
+            self.operation = operation
+        }
+    }
+    
+    private(set) var root: _IndexedOperationLinkedList.Node? = nil
+    private(set) weak var tail: _IndexedOperationLinkedList.Node? = nil
+    private(set) var count: Int = 0
+    private var nodeForOperation: [Operation: _IndexedOperationLinkedList.Node] = [:]
+    
+    func insert(_ operation: Operation) {
+        let node = _IndexedOperationLinkedList.Node(operation: operation)
+        
+        if root == nil {
+            root = node
+            tail = node
+        } else {
+            node.previous = tail
+            tail?.next = node
+            tail = node
+        }
+        nodeForOperation[node.operation] = node
+        count += 1
+    }
+    
+    func remove(_ operation: Operation) {
+        guard let node = nodeForOperation.removeValue(forKey: operation) else {
+            return
+        }
+        
+        guard let unwrappedRoot = root, let unwrappedTail = tail else {
+            // There cannot be a case where `nodeForOperation` contains a node and either `root` or `tail` is nil
+            // When root and tail are `nil`, `nodeForOperation` dictionary must be empty.
+            fatalError()
+        }
+        
+        if node === unwrappedRoot {
+            let next = node.next
+            next?.previous = nil
+            root = next
+            count -= 1
+            return
+        }
```

Comment:
from a readability standpoint it might be nicer to have these read as:
```
if node === unwrappedRoot {
...
} else if node === unwrappedTail {
...
} else {
...
}
```

not absolutely needed but might be nice for making it super clear.

Reply:
I think using of **if-elseif-else** improves the code readability. I will update the code.

Reply:
Resolving this comment since it is addressed